### PR TITLE
Remove SQLConnectionHandler

### PR DESCRIPTION
**Pull request by squirrelo**
This updates the survey to use the Transaction object. Since that was the last thing using the SQLConnectionHandler for database access, it also removes that as well.

Requires #535

### CODE REVIEW

File: amgut/lib/data_access/survey.py [hunk 1]
```diff
@@ -34,51 +32,56 @@ class Question(object):
...
+                         ON qr.response = sr.{0}
+                     WHERE q.survey_question_id = %s
+                     ORDER BY qr.display_index
+                  """.format(_LOCALE_COLUMN, self._survey_question_table,
+                             self._question_response_table,
+                             self._response_table)
+            TRN.add(sql, [self.id])
+            self.responses = TRN.execute_fetchflatten()
+
+            if not self.responses:
```
Comment (mortonjt) [inline amgut/lib/data_access/survey.py]: What sort of things would trigger this call?

Comment (squirrelo) [inline amgut/lib/data_access/survey.py]: String or Text question type which doesn't have set responses

**Comment by mortonjt** [comment]
Don't forget to resolve the merge conflicts

**Comment by squirrelo** [comment]
Resolved, good to review

File: amgut/lib/data_access/survey.py [hunk 2]
```diff
@@ -282,50 +294,46 @@ def fetch_survey(self, survey_id):
...
+            answers_other = TRN.execute_fetchindex()
+
+            survey = defaultdict(list)
+            for qid, idx, qtype in answers:
+                eid = self.questions[qid].interface_element_ids[0]
+                if qtype == 'SINGLE':
+                    survey[eid] = idx
+                else:
+                    survey[eid].append(idx)
+
```
Comment (mortonjt) [inline amgut/lib/data_access/survey.py]: This is another personal preference

I'm not a big fan of hardcoding numbers.  Is it possible to replace these numbers with constants, or add a comment indicating why those indices are chosen.

Comment (squirrelo) [inline amgut/lib/data_access/survey.py]: Byproduct of the Transaction object, so can't be changed.

Comment (mortonjt) [inline amgut/lib/data_access/survey.py]: Sorry, was referring to line 330

File: amgut/lib/data_access/survey.py [hunk 3]
```diff
@@ -282,52 +294,46 @@ def fetch_survey(self, survey_id):
...
+            for qid, data in answers_other:
+                eid = self.questions[qid].interface_element_ids[0]
+                data = str(data[2:-2])  # drop [""]
```
Comment (mortonjt) [inline amgut/lib/data_access/survey.py]: Here's what I was referring to

**Comment by mortonjt** [comment]
One small comment, but otherwise looks good

File: amgut/lib/data_access/survey.py (hunk 3 above)
Comment (squirrelo) [inline amgut/lib/data_access/survey.py]: There is a comment at the end of the line explaining what it does? I'm not sure what else I can do.

Comment (squirrelo) [inline amgut/lib/data_access/survey.py]: Also note I did not program that, so it's been there since the beginning.

Comment (squirrelo) [inline amgut/lib/data_access/survey.py]: although looking at that, I think there is a cleaner way to do it. Will futz with it in a bit.

**Comment by squirrelo** [comment]
Comment addressed, good to go.

**Comment by mortonjt** [comment]
:+1:

**Pull request merged**